app/components/at_mention/at_mention.test.js

```diff
@@ -0,0 +1,157 @@
+    });
+
+    test('should match snapshot, for user without username', () => {
+        const usersByUsername = {'user-1': {id: 'user-1'}};
+        const wrapper = shallow(
+            <AtMention
+                {...baseProps}
+                usersByUsername={usersByUsername}
+            />
+        );
+
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
+
+    test('should match snapshot, with suffix', () => {
+        const usersByUsername = {'user-1': {id: 'user-1', username: 'user'}};
+        const wrapper = shallow(
+            <AtMention
+                {...baseProps}
+                usersByUsername={usersByUsername}
+            />
+        );
+
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
+
+    test('should call Clipboard.setString on handleCopyMention', () => {
+        Clipboard.setString = jest.fn();
+        const wrapper = shallow(
+            <AtMention {...baseProps}/>
+        );
+
+        const username = 'user-1';
+        wrapper.setState({username});
+        wrapper.instance().handleCopyMention();
+        expect(Clipboard.setString).toHaveBeenCalledTimes(1);
+        expect(Clipboard.setString).toBeCalledWith(`@${username}`);
+    });
+
+    test('should call props.onLongPress and mattermostManaged.getLocalConfig on handleLongPress', async () => {
+        const translationId = {
+            id: 'mobile.mention.copy_mention',
+            defaultMessage: 'Copy Mention',
+        };
+        const formatMessage = jest.fn().mockReturnValue(translationId);
+        mattermostManaged.getLocalConfig = jest.fn().
+            mockResolvedValueOnce({copyAndPasteProtection: 'false'}).
+            mockResolvedValue({copyAndPasteProtection: 'true'});
+        const onLongPress = jest.fn();
+        const wrapper = shallow(
+            <AtMention
+                {...baseProps}
+                onLongPress={onLongPress}
+            />, {context: {intl: {formatMessage}}},
+        );
+
+        wrapper.instance().handleCopyMention = jest.fn();
+
+        // if not allowed to copy
+        await wrapper.instance().handleLongPress();
+        expect(mattermostManaged.getLocalConfig).toHaveBeenCalledTimes(1);
+        expect(onLongPress).toHaveBeenCalledTimes(1);
+        expect(onLongPress).lastCalledWith(undefined);
+
+        // if allowed to copy
+        const action = {text: translationId, onPress: wrapper.instance().handleCopyMention};
+        await wrapper.instance().handleLongPress();
+        expect(mattermostManaged.getLocalConfig).toHaveBeenCalledTimes(2);
+        expect(onLongPress).toHaveBeenCalledTimes(2);
+        expect(onLongPress).lastCalledWith(action);
+    });
+
+    test('should match return value on getUserDetailsFromMentionName', () => {
+        const wrapper = shallow(
+            <AtMention {...baseProps}/>
+        );
+
+        expect(wrapper.instance().getUserDetailsFromMentionName(baseProps)).toEqual({id: 'user-1', username: 'username_1'});
```

Comment:
Only one test case?

Reply:
Added test cases.  Also change arguments of `getUserDetailsFromMentionName` so it can be tested easily.